### PR TITLE
[JENKINS-55582] Turn modules into plugins

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-55582](https://issues.jenkins-ci.org/browse/JENKINS-55582)

Supersedes #18 & #19.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -11,8 +11,8 @@
 
   <groupId>org.jenkins-ci.modules</groupId>
   <artifactId>instance-identity</artifactId>
-  <version>2.3-SNAPSHOT</version>
-  <packaging>jenkins-module</packaging>
+  <version>${revision}${changelist}</version>
+  <packaging>hpi</packaging>
 
   <name>Instance Identity</name>
   <description>Maintains an RSA key pair that can serve as a foundation of authentication when communicating with Jenkins</description>
```
Comment (jglick) [inline pom.xml]: This and the other modules will need `url`s and their own wiki pages.

**Comment by timja** [comment]
anything left on this?

**Comment by jglick** [comment]
Looking at https://github.com/jenkinsci/jep/pull/326 to refresh my memory…

**Review by timja — APPROVED** [review]

**Comment by jglick** [comment]
@timja do you feel like releasing this and integrating into core? Should be coördinated somehow with the `sshd` change.

**Comment by timja** [comment]
sure, I created this for an incremental so wait for that first I guess? https://github.com/jenkinsci/jenkins/pull/5303

**Comment by timja** [comment]
https://github.com/jenkinsci/instance-identity-plugin/releases/tag/instance-identity-3.0

@jglick you got time to create the core PR?

**Comment by jglick** [comment]
Sure I can try that.

**Comment by jglick** [comment]
Working on it. Tricky due to split plugin cycles.

**Pull request merged**